src/App.tsx

```diff
@@ -1,61 +1,135 @@
+import { ChangeEvent, useState } from 'react';
+import { TodoList } from './components/TodoList';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+const getUserById = (userId: number): User | null => (
+  usersFromServer.find(user => user.id === userId) || null
+);
+
+const getUserByName = (name: string): User | null => (
+  usersFromServer.find(user => user.name === name) || null
+);
+
+const refactoredTodos: Todo[] = todosFromServer.map((todo) => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [todos, setTodos] = useState(refactoredTodos);
+  const [title, setTitle] = useState('');
+  const [selectedUser, setSelectedUser] = useState('');
+  const [titleValid, setTitleValid] = useState(true);
+  const [userValid, setUserValid] = useState(true);
+
+  const handleTitleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setTitleValid(Boolean(event.target.value));
+  };
+
+  const handleUserChange = (event: ChangeEvent<HTMLSelectElement>) => {
+    setSelectedUser(event.target.value);
+    setUserValid(Boolean(event.target.value));
+  };
+
+  const handleFormSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    setTitleValid(Boolean(title));
+    setUserValid(Boolean(selectedUser));
+
+    if (!selectedUser || !title) {
+      return;
+    }
+
+    setTodos((currentTodos) => {
+      const maxId = Math.max(...currentTodos.map(todo => todo.id));
+      const newUser = getUserByName(selectedUser);
+
+      return [
+        ...currentTodos,
+        {
+          id: maxId + 1,
+          title,
+          completed: false,
+          userId: newUser ? newUser.id : null,
+          user: newUser,
+        },
+      ];
+    });
+
+    setTitle('');
+    setSelectedUser('');
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        onSubmit={handleFormSubmit}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label>
+            {'Title: '}
+
+            <input
+              type="text"
+              data-cy="titleInput"
+              value={title}
+              onChange={handleTitleChange}
+            />
```

Comment:
You don't have placeholder in title input

![image](https://github.com/mate-academy/react_add-todo-form/assets/84296041/28d50884-c9c7-4061-8afb-21c3a762c738)
![image](https://github.com/mate-academy/react_add-todo-form/assets/84296041/f9b7dcd1-f345-42b7-8e73-f8ea59bdadcb)